### PR TITLE
refactor output in case of error for async callback and empty function name check

### PR DESCRIPTION
**Pull request by laurci**

### CODE REVIEW

File: vmhost/contexts/output.go [hunk 1]
```diff
@@ -562,22 +562,60 @@ func (context *outputContext) DeployCode(input vmhost.CodeDeployInput) {
...
+		OutputAccounts: make(map[string]*vmcommon.OutputAccount),
+	}
+
+	err = async.DeleteFromCallID(callId)
+	if err != nil {
+		logOutput.Trace("failed to delete Async Context", "callId", callId, "err", err)
+	}
+
+	vmOutput := context.outputState // GetVMOutput updates metering
+	context.PopSetActiveState()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply vmhost/contexts/output.go:588]: no need to pop here.

File: vmhost/flags.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package vmhost
 import "github.com/multiversx/mx-chain-core-go/core"
 
 const (
+	// AsyncV3Flag defines the flag that activates async v3
+	AsyncV3Flag core.EnableEpochFlag = "AsyncV3Flag"
```
Comment (sasurobert) [inline vmhost/flags.go:7]: this has to be the last.

Comment (sasurobert) [reply vmhost/flags.go:7]: also I made a cleanup somewhere of these flags, as they are not needed. Next week we need to make a merge from a new branch to feat/asyncV3

File: vmhost/contexts/output.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (laurci) [reply vmhost/contexts/output.go]: I swapped the order. `AsyncV3Flag` is now last.

**Review by sasurobert — APPROVED** [review]

**Pull request merged**